projects/core/src/auth/user-auth/facade/auth.service.ts

```diff
@@ -52,6 +52,10 @@ export class AuthService {
    */
   async checkOAuthParamsInUrl(): Promise<void> {
     try {
+      if (await this.authMultisiteIsolationService?.isDifferentBaseSite()) {
+        this.logout();
```

Comment:
I'm just wondering that if we log out after switching a base site, our credentials would be lost and we'd need to log in again if we switched back immediately after. Is this actually what happens? 
Is it in the scope to maintain sessions between basesites?

Reply:
> I'm just wondering that if we log out after switching a base site, our credentials would be lost and we'd need to log in again if we switched back immediately after. Is this actually what happens?

Actually yes, but only if you switch from non-isolated into isolated basesite. If you switch from `electronics-spa` to `powertools-spa` the session can be shared between basesite in my opinion (as it works currently).

> Is it in the scope to maintain sessions between basesites?
- Only if you switch from non-isolated to another non-isolated, as far as I know.
There is no specific requirement for how it should work, but `isDifferentBaseSite` method runs only for isolated baseSites. So if there is any active session on non-isolated baseSite the customer should be always logged out to not mix up the sessions between isolated and non-isolated base sites. Because isolated baseSites contains a different database of users.